fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogStore.kt

```diff
@@ -0,0 +1,284 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.fenix.translations
+
+import mozilla.components.browser.state.action.TranslationsAction
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.translate.Language
+import mozilla.components.concept.engine.translate.TranslationError
+import mozilla.components.lib.state.Action
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.State
+import mozilla.components.lib.state.Store
+
+/**
+ * The [TranslationsDialogStore] holds the [TranslationsDialogState] (state tree).
+ *
+ * The only way to change the [TranslationsDialogState] inside
+ * [TranslationsDialogStore] is to dispatch an [Action] on it.
+ */
+class TranslationsDialogStore(
+    initialState: TranslationsDialogState,
+    middlewares: List<Middleware<TranslationsDialogState, TranslationsDialogAction>> = emptyList(),
+) : Store<TranslationsDialogState, TranslationsDialogAction>(
+    initialState,
+    TranslationsDialogReducer::reduce,
+    middlewares,
+) {
+    init {
+        dispatch(TranslationsDialogAction.FetchSupportedLanguages)
+    }
+}
+
+/**
+ * The current state of the Translations bottom sheet dialog.
+ *
+ * @property isTranslated The page is currently translated.
+ * @property isInProgress The page is currently attempting a translation.
```

Comment:
nit: Slight recommendation for renaming `isInProgress` to `translationInProgress` or similar. Reason for recommendation is a translation restore can also be in progress and this value appears only tied to a translation in progress.

Reply:
Done